businessRules/donations.js

```diff
@@ -1,54 +1,59 @@
 const { v4 } = require('uuid');
 const { createDonation } = require('../models/donations');
-const { getAllIdeas } = require('../models/ideas');
-const { getIdeas } = require('../utils/storage');
+const { validateIdea } = require('../validations/ideas');
+const { validateSum } = require('../validations/donations.js');
+const { getUser } = require('./users');
+const { getIdea } = require('./ideas');
 
-function createDonation(firstName, sum, userId, ideaId) {
-    let ideas = getIdeas();
-    const idea = ideas.find((idea) => ideaId === idea.id);
+function createAnonymousUser(firstName, userId) {
+    userId = v4();
+    const anonymous = 'Anonymous';
+
+    firstName = firstName || anonymous;
+}
 
-    if (!idea) {
-        throw new Error('Error. No idea with such ID found.');
+function createDonationByUnregisteredUser(sum, firstName, ideaId) {
+    validateIdea(ideaId);
+    validateSum(sum);
+    let userId;
+    if (!firstName) {
+        createAnonymousUser(firstName, userId);
```

Comment:
You're passing `userId` to `createAnonymousUser`, but `userId` is `undefined`

Reply:
I've checked that function, and decided not to provide that, because user is unreg. and he physically doesn't have user.id so it will be generated in createAnonymousUser